### ISSUE TITLE
Allow specification of sidebar position in frontmatter of each dir/file

### ISSUE THREAD

**Issue by jakerenzella** [withastro/starlight#524]
### What version of `starlight` are you using?

0.6.1

### What is your idea?

We are working on a longer form book: https://programmers.guide/book/part-0-getting-started/00-introduction/

We would like to specift the order/position in the sidebar for each menu item, as well as overriding the name of the sidebar title in in frontmatter of each dir/file so we wouldn't have to do it all centrally from the config file.

### Why is this feature necessary?

With a large number of directories and files, managing centrally in the config file can become messy. Simply including a position index position in the frontmatter makes the ordering simpler for our use case

### Do you have examples of this feature in other projects?

https://docusaurus.io/docs/next/sidebar/autogenerated#category-item-metadata

### Participation

- [X] I am willing to submit a pull request for this issue.

**Comment by jakerenzella** [comment]
FYI I think this is similar/would be solved by: https://github.com/withastro/starlight/issues/370 as that would allow us to configure the directories via a config file in situ

**Comment by haikyuu** [comment]
Is there a workaround I can use for now?

**Comment by lschierer** [comment]
yeah, solving [#370](https://github.com/withastro/starlight/issues/370) would allow this submitter to position the entire directory via the frontmatter for the directory index pages and the sortOrder would then operate within that directory.  I think that's what the submitter is describing as desired state.
